### PR TITLE
doc: Update philosophy document

### PR DESCRIPTION
**Pull request by youngnick**
This commit updates the Contour philosophy document, as our position
on how we make settings configurable hss evolved since it was first
written.

Also, gets rid of those stupid smart quotes.

Signed-off-by: Nick Young <ynick@vmware.com>

### CODE REVIEW

**Comment by youngnick** [comment]
So, I've made this update to the philosophy document. I *think* that it reflects where we are at now, but I'm interested in other thoughts. Anyone?

File: site/_resources/philosophy.md [hunk 1]
```diff
@@ -51,17 +52,23 @@ For example, many projects allow network buffer sizes to be configured arbitrari
 In providing the tools, but little or no guidance on how they should be used, infinitely tweakable knobs forces the burden of mastering the high dimensional configuration space onto the end users of the product.
 The secondary impact on support teams and the upstream development team should not be discounted, both in terms of difficulty in diagnosing problems, and the possibility of configuration drift between environments.
 
-Contour takes the position that when there is a sensible default value for an Envoy configuration parameters, Contour will apply it unconditionally.
-We’ve used this position in the past by compressing HTTP response bodies unconditionally, disallowing TLS/1.0, choosing an aggressive cipher suite, and so on.
+Contour takes the position that when there is a sensible default value for an Envoy configuration parameters, Contour will apply.
```
Comment (skriss) [inline site/_resources/philosophy.md:55]:
```suggestion
Contour takes the position that when there is a sensible default value for an Envoy configuration parameters, Contour will apply it.
```
```suggestion
Contour takes the position that when there is a sensible default value for an Envoy configuration parameters, Contour will apply.
```

File: site/_resources/philosophy.md [hunk 2]
```diff
@@ -51,17 +52,23 @@ For example, many projects allow network buffer sizes to be configured arbitrari
...
+When Contour was first developed, generally we stopped there.
+However, users of Contour found that this approach, while excellent for the combined cluster operator/application developer users, was not ideal when the cluster operator had to keep a large production deployment up and running.
+
+Contour still and always will attempt to choose the best default for the largest set of users, but we will allow tweaking of those defaults when requested.
+The TLS cipher suite is a great example of this.
+Contour will move reasonably quickly on keeping the default cipher suite up to date with the current standards, but we allow configuration of the cipher suite, so cluster operators have the option to override the defaults as they need to.
+
+In addition, sometimes it's useful for application developers to be able to override some global defaults.
+When we do allow overriding of defaults, we will do our best to also supply [Gatekeeper](https://github.com/open-policy-agent/gatekeeper) policies and validations to allow cluster operators to limit the *range* of values that app devs may configure.
```
Comment (skriss) [inline site/_resources/philosophy.md:66]: we're probably due for doing some more work here :)

**Review by skriss — APPROVED** [review]
Sounds pretty reasonable to me. A couple minor non-blocking comments.

File: site/_resources/philosophy.md (hunk 2 above)
Comment (youngnick) [reply site/_resources/philosophy.md:66]: Yeah, I thought the same thing when I looked at the directory.

File: site/_resources/philosophy.md [hunk 3]
```diff
@@ -51,17 +52,23 @@ For example, many projects allow network buffer sizes to be configured arbitrari
...
 
 ### Limited feature scope
 
 The Contour project has a well defined scope: a reverse proxy implementation for HTTP workloads on Kubernetes clusters.
-Contour’s limited support for TCP proxying is intended solely for Contour to to support web applications which desire to handle TLS directly.
+Contour's limited support for TCP proxying is intended solely for Contour to to support web applications which desire to handle TLS directly.
```
Comment (stevesloka) [inline site/_resources/philosophy.md:71]: There's a double `to` in the sentence.

File: site/_resources/philosophy.md [hunk 4]
```diff
@@ -51,17 +52,23 @@ For example, many projects allow network buffer sizes to be configured arbitrari
...
+We've used this position in the past by compressing HTTP response bodies unconditionally, disallowing TLS/1.0 and 1.1, choosing as secure as possible a default cipher suite, and so on.
 
-Contour will provide a method for the administrator or developer to supply their chosen value as a last resort, in the event that we cannot discover a universally acceptable value.
-In the history of the project this second scenario has rarely occurred.
-More often than not, the discussion sparked by the desire to change a particular parameter has led to a deeper understanding of the ways in which Contour is being used which would not have otherwise occurred.
+When Contour was first developed, generally we stopped there.
+However, users of Contour found that this approach, while excellent for the combined cluster operator/application developer users, was not ideal when the cluster operator had to keep a large production deployment up and running.
+
+Contour still and always will attempt to choose the best default for the largest set of users, but we will allow tweaking of those defaults when requested.
+The TLS cipher suite is a great example of this.
```
Comment (stevesloka) [inline site/_resources/philosophy.md:62]: I don't think we need the TLS cipher suite example here, the paragraph before should be enough to keep the doc smaller.

**Review by stevesloka — CHANGES_REQUESTED** [review]

File: site/_resources/philosophy.md (hunk 4 above)
Comment (youngnick) [reply site/_resources/philosophy.md:62]: Sure, sounds good.

**Review by stevesloka — APPROVED** [review]

**Pull request merged**